src/Proto.Actor/Context.cs

```diff
@@ -212,6 +218,23 @@ public void Unwatch(PID who)
 
         public async Task InvokeUserMessageAsync(object msg)
         {
+            var influenceTimeout = true;
+            if (ReceiveTimeout > TimeSpan.Zero)
+            {
+                var notInfluenceTimeout = msg is INotInfluenceReceiveTimeout;
+                influenceTimeout = !notInfluenceTimeout;
+                if (influenceTimeout)
+                    StopReceiveTimeout();
+            }
+
+            await ProcessMessageAsync(msg);
```

Comment:
There's a bit of work to be done to get the message invocation in line with Go. Parts of it are at least extracted to a ProcessMessageAsync method, but there are some major differences left, e.g. the receive pipeline.